json/test/src/fs2/data/json/mergepatch/JsonMergePatchTest.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2019 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fs2.data.json
+package mergepatch
+
+import circe._
+
+import cats.effect._
+
+import diffson.jsonmergepatch.JsonMergePatch
+import diffson.circe._
+
+import io.circe._
+
+import fs2._
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class JsonMergePatchTest extends AnyFlatSpec with Matchers {
+
+  val valuePatch = JsonMergePatch.Value(Json.fromInt(5))
+  val nullPatch = JsonMergePatch.Value(Json.Null)
+  val objectPatch = JsonMergePatch.Object(
+    Map("key1" -> Json.fromInt(3),
+        "key2" -> Json.obj("nested" -> Json.fromString("s"), "new" -> Json.True, "deleted" -> Json.Null)))
+  val objectPatchTokens = List(
```

Comment:
Maybe something slightly more readable? Like
```scala
val objectPatchTokens = CirceTokenizer.tokenize(
    Json.obj(
      "key1" := 3,
      "key2" := Json.obj(
        "nested" := "s",
        "new" := true,
        "deleted" := Json.Null
      )
    )
  ).toList
```

Reply:
I can even use `parse` :)

Reply:
After trying, actually nope it doesn't work as the circe object field map has arbitrary order for keys, whereas the stream tokens are ordered. So sadly, I can't make it more readable.